mono/mini/TestDriver.cs

```diff
@@ -13,10 +13,40 @@ public class CategoryAttribute : Attribute
 		get; set;
 	}
 }
+public class TestDriverReporter
+{
+	private int failedTests = 0;
+	private int skippedTests = 0;
+	private int executedTests = 0;
+
+	public int FailedTests {
+		get {
+			return failedTests;
+		}
+	}
+
+	public int SkippedTests {
+		get {
+			return skippedTests;
+		}
+	}
+
+	public int ExecutedTests {
+		get {
+			return executedTests;
+		}
+	}
+
+	public void ReportResults (int executed, int skipped, int failed) {
+		executedTests = executed;
+		skippedTests = skipped;
+		failedTests = failed;
+	}
+};
 
 public class TestDriver {
 
-	static public int RunTests (Type type, string[] args) {
+	static private int internalRunTests(Type type, string[] args, TestDriverReporter reporter) {
```

Comment:
no need to specify `private`, it is the default. Also C# convention is to use PascalCasing for method names.